server/auth/auth.go

```diff
@@ -70,10 +75,24 @@ type Signer interface {
 	PubKey() *secp256k1.PublicKey
 }
 
-// FeeChecker is a function for retrieving the details for a fee payment.
+// FeeChecker is a function for retrieving the details for a fee payment txn.
 type FeeChecker func(assetID uint32, coinID []byte) (addr string, val uint64, confs int64, err error)
 
-type TxDataSource func([]byte) ([]byte, error)
+// BondTxParser parses a dex fidelity bond transaction and the redeem script of
+// the first output of the transaction, which must be the actual bond output.
+// The returned account ID is from the second output. This will become a
+// multi-asset checker.
+//
+// NOTE: For DCR, and possibly all assets, the bond script is reconstructed from
+// the null data output, and it is verified that the bond output pays to this
+// script. As such, the provided bondData (redeem script for UTXO assets) is not
+// used. Maybe remove it from the function signature if no asset requires any
+// additional data.
+type BondTxParser func(assetID uint32, version uint16, rawTx, bondData []byte) (bondCoinID []byte,
+	amt int64, bondAddr string, lockTime int64, acct account.AccountID, err error)
```

Comment:
Why should `AuthManager` care about the bond address? It seems asset-specific and unrelated to the task of validation.

Reply:
Is it just so we can `if chkAddr != bondAddr ` and also use the same `FeeChecker` interface?

Reply:
I think I meant to remove the address return.  Need to follow up now that the bond tx structure is reworked and merged.  Will get to asap.